packages/astro/src/compat/index.ts

```diff
@@ -0,0 +1,34 @@
+import type { LogOptions } from '../core/logger/core';
+import type { UserConfig as VitestConfig } from 'vitest/config';
+
+import * as vite from 'vite';
+import * as vitest from 'vitest';
+import { nodeLogDestination } from '../core/logger/node.js';
+import { openConfig } from '../core/config.js';
+import { createVite } from '../core/create-vite.js';
+import { runHookConfigDone, runHookConfigSetup } from '../integrations/index.js';
+
+export async function getViteConfig(): Promise<vite.UserConfig> {
+	let logging: LogOptions = {
+		dest: nodeLogDestination,
+		level: 'info',
+	};
+	let { astroConfig: config } = await openConfig({
+		cmd: 'test',
+		logging,
+	});
+	await runHookConfigSetup({ config, command: 'dev' });
+	const viteConfig = await createVite(
+		{
+			mode: 'test',
+		},
+		{ astroConfig: config, logging: logging, mode: 'dev' }
+	);
+	await runHookConfigDone({ config });
+	return viteConfig;
+}
+
+export async function getVitestConfig(test: vitest.UserConfig = {}): Promise<VitestConfig> {
+	const viteConfig = await getViteConfig();
+	return Object.assign(viteConfig, { test });
```

Comment:
Should this be `Object.assign(viteConfig, test);` instead of `Object.assign(viteConfig, { test });` ?

Reply:
Updated this signature to just accept an inline config with overrides and it is merged in.